### PR TITLE
[ci] #3562: Fix PR-generator Jenkinsfile & Add Coverall badge & Set up JDK 11

### PR DESCRIPTION
**Pull request by BAStos525**
## Description
1. Change `README.MD` badge from `Codecov` to `Coveralls` tool.
2. Update PR-generator `Jenkinsfile`.
3. Add `docker.soramitsu.co.jp/iroha2/iroha2:dev-${{ github.event.pull_request.head.sha }}` iroha2 image tag to push to Soramitsu registry for PR-generator.
4. Add JDK distribution to `Set up JDK 11` step into `java-api` job inside `I2::Release::Tests` workflow.

### Linked issue
1. We don't have `Coveralls` coverage tool badge.
2. PR-generator deployment hasn't bee finished.
3. #3562 

### Benefits
1. `Coveralls` badge.
2. PR-generator should work after merging the corresponding PR with Jenkins library (I'LL NOTIFY WHEN IT WILL BE ABLE TO BE TESTED)!
3. Fix JDK 11 Set up in the `Release-PR` workflow.

### Checklist

- [X] Done the work
- [X] CI checks pass
- [X] Commits are good
- [ ] Addressed comments
- [ ] Tested badge as UI element.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
-def pipeline = new org.iroha2.AppPipeline(steps: this,
+def pipeline = new org.iroha2PRDeploy.AppPipeline(steps: this,
     k8sPrDeploy: true,
     vaultPrPath: "argocd-cc/src/charts/iroha2/environments/tachi/",
-    vaultUser: "iroha2-ro",
+    vaultUser: "iroha2-rw",
     vaultCredId: "iroha2VaultCreds",
     valuesDestPath: "argocd-cc/src/charts/iroha2/",
-    devValuesPath: "dev/test/"
+    devValuesPath: "dev/dev/",
```
Comment (appetrosyan) [inline Jenkinsfile:9]: Why this change? `dev/dev/` looks like a typo.

File: Jenkinsfile [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
+    initialSecretName: "iroha2-eso-base",
+    initialNameSpace: "iroha2-dev",
+    targetNameSpace: "iroha2-${env.CHANGE_ID}-web",
+    targetSecretName: "iroha2-${env.CHANGE_ID}-iroha2-pr-eso-base",
+    disableSecretScanner: true
```
Comment (appetrosyan) [inline Jenkinsfile:14]: I'm guessing that we should re-enable it later, right?

File: Jenkinsfile (hunk 1 above)
Comment (BAStos525) [reply Jenkinsfile:9]: it's ArgoCD images versioning: `tag/namespace`. So, it's not a typo.

File: Jenkinsfile (hunk 2 above)
Comment (BAStos525) [reply Jenkinsfile:14]: If you mean `SecretScanner`, probably yes. I guess the problem in iroha keys. It's not a really important right now, so we can back to this later.
